### PR TITLE
expose templates

### LINKED ISSUE #11: Template full URL paths
I'd love to use the HTML templates in an external FastAPI app.  Problem is my urls are different than what is used by rio-viz, for example my tilejson url is `${api_endpoint}/cog/tilejson.json?${url_params}`.

There are only a handful of routes which are called (`point`, `tilejson`, and `metadata`) and templating the full url would allow the HTML to be more easily consumed by other applications without adding much complexity to rio-viz.

The caller would of course have to ensure the request parameters and output for a certain route are the same as rio-viz, but if your app didn't have similar requests/responses you probably wouldn't be trying to use the HTML templates to begin with.

### PR DESCRIPTION
**Pull request by geospatial-jeff**
Expose full urls in HTML templates and change how templates are created to make them easier to use in external apps.

You can import templates (`from rio_viz.templates import html_templates`) and call them as usual to generate the html.

If you are using FastAPI, you may use `from rio_viz.templates import create_template_factory` to create a FastAPI dependency which may be used to inject the templates into your application.  The factory parameters let you customize which endpoints are used in the template.

For example, if the path operation to create a `tilejson` is bound to the `create_tilejson` function you may use the following factory which will create the correct template regardless of the route defined by that operation.

```python
template_factory = create_template_factory(tilejson="create_tilejson")
```

Closes #11

### CODE REVIEW

**Comment by vincentsarago** [comment]
thanks @geospatial-jeff this is really nice. 

Would you be able to add an example in the Readme of how we can expand rio-viz using this solution ?

**Comment by geospatial-jeff** [comment]
I split the factory into two, one per template.  I think the usage is better this way because each template takes different parameters and it avoid having to specify the name of the function consuming the dependency.

File: rio_viz/templates/template.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+from typing import Callable, Optional
+
+from starlette.requests import Request
+from starlette.templating import Jinja2Templates, _TemplateResponse
+
+html_templates = Jinja2Templates(directory=os.path.dirname(__file__))
+
+
+def create_simple_template_factory(
+    tilejson: Optional[str] = "_tilejson", info: Optional[str] = "_info"
```
Comment (vincentsarago) [inline rio_viz/templates/template.py:11]: @geospatial-jeff do we have to set `Optional` here, because there is a default value I'm not sure we need this?

**Comment by vincentsarago** [comment]
@geospatial-jeff thanks for the change, I'm going to take this over and merge with change I just pushed to master.

File: rio_viz/templates/template.py (hunk 1 above)
Comment (vincentsarago) [reply rio_viz/templates/template.py:11]: @geospatial-jeff second question, is `_tilejson` indicating the name of the function for the `@self.app.get("/tilejson.json")` ? What kind of black magic this is ?

**Pull request closed without merging**